File: cyberduck/security_txt.py
"""Parser for security.txt style contact files."""

from __future__ import annotations


def parse_security_txt(text: str) -> dict[str, list[str]]:
    values: dict[str, list[str]] = {}
    for line in text.splitlines():
        clean = line.strip()
        if not clean or clean.startswith("#") or ":" not in clean:
            continue
        key, value = clean.split(":", 1)
        values.setdefault(key.strip().lower(), []).append(value.strip())
    return values


def has_contact(text: str) -> bool:
    return "contact" in parse_security_txt(text)

File: ts/src/defensive/securityTxt.ts
export type SecurityTxtValues = Record<string, string[]>;

/**
 * Parse a security.txt-like file.
 * Defensive purpose: extraction only (no network calls, no scanning).
 */
export function parseSecurityTxt(text: string): SecurityTxtValues {
  const values: SecurityTxtValues = {};

  for (const rawLine of text.split(/\r?\n/)) {
    const line = rawLine.trim();
    if (!line || line.startsWith('#') || !line.includes(':')) continue;
    const [k, ...rest] = line.split(':');
    const key = k.trim().toLowerCase();
    const value = rest.join(':').trim();
    if (!key) continue;
    if (!values[key]) values[key] = [];
    values[key].push(value);
  }

  return values;
}

export function hasContact(text: string): boolean {
  const parsed = parseSecurityTxt(text);
  return Array.isArray(parsed['contact']) && parsed['contact'].length > 0;
}

